### ISSUE TITLE
Achievements page crashes in  Firefox 102.10.0esr

### ISSUE THREAD

**Issue by UlyC** [Crossbell-Box/xLog#556]
`Application error: a client-side exception has occurred (see the browser console for more information).`

```
TypeError: l.items.findLastIndex is not a function
    v AchievementModal.tsx:149
    rs react-dom.production.min.js:108
    iv react-dom.production.min.js:263
    oR react-dom.production.min.js:249
    oD react-dom.production.min.js:247
    oD react-dom.production.min.js:247
    ow react-dom.production.min.js:242
    ok react-dom.production.min.js:240
    R scheduler.production.min.js:14
    A scheduler.production.min.js:15
    97274 scheduler.production.min.js:15
    Webpack 31
app-index.js:32:21
    error app-index.js:32
    ln react-dom.production.min.js:139
    callback react-dom.production.min.js:140
    nf react-dom.production.min.js:83
    nd react-dom.production.min.js:83
    av react-dom.production.min.js:201
    ak react-dom.production.min.js:202
    aD react-dom.production.min.js:224
    ak react-dom.production.min.js:202
    aD react-dom.production.min.js:224
    ak react-dom.production.min.js:202
    aD react-dom.production.min.js:224
    ak react-dom.production.min.js:202
    aD react-dom.production.min.js:224
    ak react-dom.production.min.js:202
    aD react-dom.production.min.js:224
    ak react-dom.production.min.js:202
    aD react-dom.production.min.js:224
    ak react-dom.production.min.js:203
    aD react-dom.production.min.js:224
    ak react-dom.production.min.js:202
    oU react-dom.production.min.js:257
    oU react-dom.production.min.js:255
    oE react-dom.production.min.js:242
    ok react-dom.production.min.js:241
    R scheduler.production.min.js:14
    A scheduler.production.min.js:15
    (Async: EventHandlerNonNull)
    97274 scheduler.production.min.js:15
    Webpack 31
```

![image](https://github.com/Crossbell-Box/xLog/assets/36838240/880945db-e610-45e3-9727-f281435e51f7)

**Comment by enpitsuLin** [comment]
it seems because `Array.prototype.findLastIndex()` not support on firefox 104-

maybe polyfill or do reverse then use `findIndex()`

**Comment by UlyC** [comment]
> but firefox 104 was already released 1 yr ago, how about upgrading your browser first?

Version 102.10.0, first offered to ESR channel users on April 11, 2023

**Comment by enpitsuLin** [comment]
> > but firefox 104 was already released 1 yr ago, how about upgrading your browser first?
> 
> Version 102.10.0, first offered to ESR channel users on April 11, 2023

my bad, I think this problem really needs to be solved

**Comment by Georgehu716** [comment]
I encounter the same error when visit https://hellotangman.xlog.app, both in firefox (115.0) and chrome (114.0.5735.198 Official Build arm64)
![Screenshot 2023-07-08 at 16 58 05](https://github.com/Crossbell-Box/xLog/assets/26440011/f072d8a8-1b22-4611-acc0-745af5831745)

**Comment by DIYgod** [comment]
Fixed in cfcc6c562b42a7ee193f937d4318307421595bed
